[Moloco-RMP] Rename RMP to MCM, Restoring test cases

heonjang opened the pull request:
## Testing

- [✅ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [✅ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)

## Description

1. Moloco RMP changed its name to Moloco MCM. Change the names in the code accordingly.
2. Restore removed breaking changes

- https://github.com/segmentio/action-destinations/commit/96ef3001e634001398a06b8894882aa20c3e8420
![Screenshot 2024-03-26 at 4 09 42 PM](https://github.com/moloco/rmp-cdp/assets/103418311/f9b17e70-9963-40c0-a1e5-219c04dba47a)
- https://github.com/segmentio/action-destinations/commit/25e96ae139b257e0ff236cc5b0ffd47df4e255eb
![Screenshot 2024-03-26 at 4 09 16 PM](https://github.com/moloco/rmp-cdp/assets/103418311/c6fd38b8-832c-4f4c-b945-26f1e9c7f527)

> TypeError: Invalid URL

However, if this was indeed a valid error, all other tests should have failed which was not the case.
Also same exact test did not fail when retried, so I assume those were some kinds of bug at the moment.
<img width="463" alt="Screenshot 2024-03-26 at 4 12 11 PM" src="https://github.com/moloco/rmp-cdp/assets/103418311/194453c5-04e7-421e-afde-b60e13d05dfa">
<img width="503" alt="Screenshot 2024-03-26 at 4 13 06 PM" src="https://github.com/moloco/rmp-cdp/assets/103418311/002aa87a-5ff7-4db8-bc6a-8e18f51e825c">

heonjang commented:
@joe-ayoub-segment Hi Joe, it took a while to come back to you. Please take a look at the changes when you are available. Thank you in advance!

heonjang commented on packages/destination-actions/src/destinations/moloco-rmp/__tests__/snapshot.test.ts:
```diff
@@ -0,0 +1,77 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Restored snapshot test

heonjang commented on packages/destination-actions/src/destinations/moloco-rmp/addToCart/__tests__/__snapshots__/snapshot.test.ts.snap:
```diff
@@ -1,51 +1,51 @@
+exports[`Testing snapshot for MolocoMCM's addToCart destination action: all fields 1`] = `
 Object {
-  "channel_type": "2Tz2Ek0OF",
+  "channel_type": "uast)d8y]fWK6e",
```
Values changed due to the change in the `destination-slug` which is used for hashing values

heonjang commented on packages/destination-actions/src/destinations/moloco-rmp/addToWishlist/__tests__/snapshot.test.ts:
```diff
@@ -0,0 +1,75 @@
+import { createTestEvent, createTestIntegration } from '@segment/actions-core'
```
Restored snapshot test

joe-ayoub-segment commented on packages/destination-actions/src/destinations/moloco-rmp/index.ts:
```diff
@@ -19,10 +19,10 @@ import purchase from './purchase'
-  name: 'Moloco Rmp',
-  slug: 'actions-moloco-rmp',
+  name: 'Moloco MCM',
+  slug: 'actions-moloco-mcm',
```
hi @heonjang - we can't change the slug. However the name change is OK. The slug isn't visible to customers other than in the URL.

heonjang commented:
> Thanks @heonjang - looks good apart from the slug renaming which we can't support. Please see comment. Please let me know if that's OK?

Hi @joe-ayoub-segment, that is fine. I've reverted that change!

joe-ayoub-segment commented:
hi @heonjang - looks like there are a couple of failing tests for your integration. Could you address these please?

Also there's a failing test because of another Integration - if you rebase from main this should be fixed. 

Best regards,
Joe

heonjang commented:
Hi @joe-ayoub-segment,

I've updated the branch up to date with the `main`.

Also, I checked that both

1. unit tests
<img width="963" alt="Screenshot 2024-04-04 at 8 19 07 PM" src="https://github.com/segmentio/action-destinations/assets/103418311/0bc715be-fe9f-47d5-a718-5255aca74fc3">

2. local server tests
<img width="1914" alt="Screenshot 2024-04-04 at 8 18 05 PM" src="https://github.com/segmentio/action-destinations/assets/103418311/c516d4dd-e741-4840-9fdd-09b80d3a2463">

passed.

It seems like CI can be run after approval from you. Please take a look whenever you are available!

joe-ayoub-segment commented:
hi @heonjang - I've rerun CI checks and there still seems to be some sort of issue. 
Here's a screenshot from the log: 
![image](https://github.com/segmentio/action-destinations/assets/45374896/47474828-3c3f-4657-baab-cadd2f8d343d)

heonjang commented on packages/destination-actions/src/destinations/moloco-rmp/__tests__/snapshot.test.ts:
```diff
@@ -0,0 +1,82 @@
+      const seedName = `${destinationSlug}#${actionSlug}`
+      const action = destination.actions[actionSlug]
+      const [eventData, _] = generateTestData(seedName, destination, action, true)
+      const settingsData = {
```
Use a constant for settingsData

heonjang commented:
Hi @joe-ayoub-segment 

Somehow, the snapshot test succeeded in a local environment but failed in the Segment's CI.

![image](https://github.com/moloco/rmp-cdp/assets/103418311/9831e9f6-e6d9-46d9-bc53-2fce5222eb44)

However, this error doesn't make sense because other actions (HOME, PURCHASE, ...) did not fail even though they share the same logic.

One possibility is that when creating a `settingsData`, the test code creates mock data by hashing the action. So somehow the LAND's result contained an invalid `platform` value which is used to create a target URL.
```
const [eventData, settingsData] = generateTestData(seedName, destination, action, true)
```

To avoid this issue, I've updated the `settingsData` so that it always contains valid data.

heonjang commented:
> Hi @joe-ayoub-segment
> 
> Somehow, the snapshot test succeeded in a local environment but failed in the Segment's CI.
> 
> ![image](https://private-user-images.githubusercontent.com/103418311/320647276-9831e9f6-e6d9-46d9-bc53-2fce5222eb44.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI2MjM3OTYsIm5iZiI6MTcxMjYyMzQ5NiwicGF0aCI6Ii8xMDM0MTgzMTEvMzIwNjQ3Mjc2LTk4MzFlOWY2LWU2ZDktNDZkOS1iYzUzLTJmY2U1MjIyZWI0NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwOVQwMDQ0NTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03N2IyNWQ0YTQwZDhjNzI4ZDQ0MjQxZTNlMDljNDVhMGM2N2RiNzRhZTc3MDkwY2IxODk1YTk0MGE1Y2YwNDJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.QyflMckLBduPnHTq_f6gSF2Qc9mvjgoExhSdVrHzIMY)
> 
> However, this error doesn't make sense because other actions (HOME, PURCHASE, ...) did not fail even though they share the same logic.
> 
> One possibility is that when creating a `settingsData`, the test code creates mock data by hashing the action. So somehow the LAND's result contained an invalid `platform` value which is used to create a target URL.
> 
> ```
> const [eventData, settingsData] = generateTestData(seedName, destination, action, true)
> ```
> 
> To avoid this issue, I've updated the `settingsData` so that it always contains valid data. ([#1953 (comment)](https://github.com/segmentio/action-destinations/pull/1953#discussion_r1556657672))

Actually,  I just removed the problematic test files so that this PR only focuses on the renaming. Let me handle the test cases separately :(

joe-ayoub-segment commented:
Hi @heonjang - the CI is still failing but I don't think it has anything to do with your code - so I'm happy to merge and deploy it next Tuesday.

joe-ayoub-segment commented:
Hi @heonjang - PR deployed. I did need to remove some tests again as they were causing issues with our CI!

joe-ayoub-segment commented:
Please confirm that you are happy with the change.

heonjang commented:
@joe-ayoub-segment Hi Joe, sorry for missing the last comment.

I checked whether the integration's name was changed or not, but it seems like the destination's name has not been updated yet.

![image](https://github.com/segmentio/action-destinations/assets/103418311/8705b687-f67b-4dc7-8927-e63b0e81286d)

Do you have an idea on this?

joe-ayoub-segment commented:
hi @heonjang that's been fixed now. 

Best regards,
Joe

heonjang commented:
> hi @heonjang that's been fixed now.
> 
> Best regards, Joe

Thanks! Confirmed that it's been fixed. Have a great rest of your week :)
